pkg/keyidentifier/testdata/make.sh

```diff
@@ -3,97 +3,170 @@
 # This script uses several tools to create ssh keys. We attempt to be
 # as exhaustive as possible to create a wide range of things to test.
 #
-# This puts them into a somewhat ugly filesystem hierarchy. The intent
-# is to group things to make it simpler for tools to test
-# groups. The layout should straightforward.
-
-
+# each function also generates a json spec file that describes various
+# attributes of a key, plus the name of the key described. This spec file is
+# consumed by golang tests, and the output of the keyidentifier package is
+# compared against these expected values.
 
 set -e
 #set -x
 
 function rand {
-    dd if=/dev/random bs=1 count=16 2> /dev/null  | base64
+    dd if=/dev/random bs=1 count=16 2> /dev/null  | base64 | sed 's/[^0-9a-zA-Z]//g'
 }
 
-# My openssh is so new, it only makes the new format
-function makeOpensshKey {
+DATA_DIR="specs"
+
+# generate some keys and the corresponding json spec files for testing fingerprints
+function makeOpensshKeyAndSpec {
     type=$1
     bits=$2
-    os="openssh7"
-    dir_fragment="$type/$bits/openssh-new"
-    mkdir -p {encrypted,plaintext}"/$dir_fragment"
-    ssh-keygen -t $type -b $bits  -C "" -f "plaintext/$dir_fragment/seph-$os" -P ""
-    ssh-keygen -t $type -b $bits  -C "" -f "encrypted/$dir_fragment/seph-$os" -P "$(rand)"
-}
+    encrypted=$3
+    format="openssh7"
+    source="openssh"
 
-makeOpensshKey rsa 1024
-makeOpensshKey rsa 2048
-makeOpensshKey rsa 4096
+    keyfile=$(rand)
+    keypath="$DATA_DIR/$keyfile"
 
-makeOpensshKey dsa 1024
 
-makeOpensshKey ecdsa 256
-makeOpensshKey ecdsa 521
+    if [ $encrypted == true ]; then
+        ssh-keygen -t $type -b $bits  -C "" -f $keypath -P "$(rand)"
+        cmd='ssh-keygen -t $type -b $bits  -C \"\" -f $keypath -P $(rand)'
```

Comment:
This duplication feels weird. Maybe bash arrays make this a little cleaner? (This pattern should be fine in all the exec places. Also, this is where I start getting sad we're using bash and not go, even though it's probably correct)

```shell
type=rsa
bits=2048
pass='""'
cmd=(ssh-keygen -t "$type" -b "$bits" -f /tmp/testkey -P "$pass")

rm -f /tmp/testkey*

eval ${cmd[*]}
echo ${cmd[*]}
echo returned $?
```

Reply:
interesting, I'm not familiar with using bash arrays like that. This works nicely